src/nix/profile.cc

```diff
@@ -329,7 +329,63 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
             manifest.elements.push_back(std::move(element));
         }
 
-        updateProfile(manifest.build(store));
+        try {
+            updateProfile(manifest.build(store));
+        } catch (BuildProfileConflictError & conflictError) {
+            std::string fallbackInstallable = "INSTALLABLE";
+            std::string conflictingInstallable = [&]() {
+                auto maybeConflictingStorePath = store->maybeParseStorePath(conflictError.conflictingPkgPath);
+                if (!maybeConflictingStorePath) {
+                    printError("warning: unable to determine installable from conflicting package path (%1%)", conflictError.conflictingPkgPath);
+                    return fallbackInstallable;
+                }
+                auto conflictingStorePath = *maybeConflictingStorePath;
+                auto conflictingManifestElement = find_if(manifest.elements.begin(), manifest.elements.end(), [&](const ProfileElement & currentProfileElement) {
+                    return currentProfileElement.storePaths.contains(conflictingStorePath);
+                });
+                if (conflictingManifestElement == manifest.elements.end()) {
+                    printError("warning: unable to find profile element for conflicting store path (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                if (!conflictingManifestElement->source) {
+                    printError("warning: conflicting store path does not have a source (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                return conflictingManifestElement->source->originalRef.to_string();
+            }();
+
+            throw Error(
+                "An existing package already provides the following file to the user's profile:\n"
```

Comment:
```suggestion
                "An existing package already provides the following file:\n"
```

@edolstra seemed a bit surprised over the verbosity of the output, and I also think we should try to keep visual noise to a minimum while offering all the assistance we can. That's what I'm trying to do in these suggestions, so don't worry about getting many of them.